data/cube/ViewRowData.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2025 Extremely Heavy Industries Inc.
+ */
+import {PlainObject, Some} from '@xh/hoist/core';
+import {flatMap} from 'lodash';
+
+/**
+ * Grouped node data, as returned by {@link Cube.executeQuery} or exposed via {@link View.result}.
+ * Designed for direct consumption by hierarchical stores and their associated tree grids.
+ */
+export class ViewRowData implements PlainObject {
+    constructor(id: string) {
+        this.id = id;
+    }
+
+    /** Unique id. */
+    id: string;
+
+    /**
+     * Label of the row. The dimension value or, for leaf rows. the underlying cubeId.
+     * Suitable for display, although apps will typically wish to customize leaf row rendering.
+     */
+    cubeLabel: string;
+
+    /** Dimension on which this row was computed, or null for leaf rows. */
+    cubeDimension: string;
+
+    /**
+     * Buckets this row appears in
+     */
+    cubeBuckets: Record<string, any>;
+
+    /**
+     * Visible children of this row.
+     *
+     * Note that this may not be the same as the simple aggregation children of this row.  In
+     * particular, this property is subject to the semantics of row locking, redundant row omission,
+     * and bucketing as defined by the Query.
+     */
+    children: ViewRowData[];
+
+    /**
+     * True for leaf rows loaded into the cube (i.e. not a grouped aggregation).
+     */
+    get isCubeLeaf(): boolean {
```

Comment:
What's the value of adding the word `Cube` in here (and `cubeDimension`, `cubeLeaves`,etc)

Reply:
just following the pattern of the existing properties with 'cubeLabel' and 'cubeDimension'.  Think the argument is to avoid naming conflicts with app properties/fields